src/factories/payments-processor-factory.ts

```diff
@@ -3,61 +3,103 @@ import { path } from 'ramda'
 
 import { createLogger } from './logger-factory'
 import { createSettings } from './settings-factory'
+import { IInvoiceRepository } from '../@types/repositories'
 import { IPaymentsProcessor } from '../@types/clients'
+import { LNbitsPaymentsProcesor } from '../payments-processors/lnbits-payment-processor'
 import { NullPaymentsProcessor } from '../payments-processors/null-payments-processor'
 import { PaymentsProcessor } from '../payments-processors/payments-procesor'
 import { Settings } from '../@types/settings'
 import { ZebedeePaymentsProcesor } from '../payments-processors/zebedee-payments-processor'
 
 const debug = createLogger('create-zebedee-payments-processor')
 
-const getConfig = (settings: Settings): CreateAxiosDefaults<any> => {
-  if (!process.env.ZEBEDEE_API_KEY) {
-    throw new Error('ZEBEDEE_API_KEY must be set.')
+export enum PaymentProcessors {
+  Zebedee = 'zebedee',
+  Lnbits = 'lnbits',
+}
+
+const validatePaymentSettings = (processor: PaymentProcessors, settings: Settings) => {
+  const callbackBaseURL = path(['paymentsProcessors', processor, 'callbackBaseURL'], settings) as string | undefined
+  if (typeof callbackBaseURL === 'undefined' || callbackBaseURL.indexOf('nostream.your-domain.com') >= 0) {
+    throw new Error(`Unable to create payments processor: Setting paymentsProcessor.${processor}.callbackBaseURL is not configured.`)
+  }
+
+  if (
+    !Array.isArray(settings.paymentsProcessors?.[processor]?.ipWhitelist)
+    || !settings.paymentsProcessors?.[processor]?.ipWhitelist?.length
+  ) {
+    throw new Error(`Unable to create payments processor: Setting paymentsProcessor.${processor}.ipWhitelist is empty.`)
+  }
+}
+
+const getPaymentProcessorConfig = (processor: PaymentProcessors, settings: Settings): CreateAxiosDefaults<any> => {
+  let apiKey: string
+
+  switch (processor) {
```

Comment:
This entire file should be split into two files:
- zebedee-payments-processor-factory.ts
- lnbits-payments-processor-factory.ts

You could then use this same file to call the factory in either of those files depending on settings.payments.processor value.

Otherwise, we need to patch this file in several spots for every new payment processor that we add.